File: Algorithms/Java/169-Majority_Element.java
class Solution {
    public int majorityElement(int[] nums) {

        int candidate = 0;
        int counter = 0;

        for(int n: nums){

            if(counter == 0){
                candidate = n;
            }

            if(candidate == n){
                counter++;
            
            }else{
                counter--;
            }

        }
        return candidate;

    }
}

File: Algorithms/Python/169-majority_element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        result = nums[0]
        count = 0

        for i in nums:
            if result == i:
                count += 1
            elif count == 0:
                result = i
            else:
                count -= 1

        return result
